File: ModSim/Lista 1/Python/FallingBall.py
class FallingBall:
  g = 9.8

  def __init__ (self):
    print "A new FallingBall object was created."
  
  def step(self):
    self.y += self.v * self.dt
    self.v -= self.g * self.dt
    self.t += self.dt
  
  def analyticPosition(self, y0, v0):
    return y0 + v0*self.t  - 0.5*self.g*self.t*self.t
  
  def analyticVelocity(self, v0):
    return v0 - self.g*self.t

File: ModSim/Lista 1/Java/FallingBall.java
public class FallingBall
{
  double y, v, t;
  double dt;
  final static double g = 9.8;
  
  public FallingBall()
  {
    System.out.println("A new FallingBall object is created.");
  } 
  public void step()
  {
    y = y + v * dt;
    v = v - g * dt;
    t = t + dt;
  }
  
  public double analyticPosition (double y0, double v0)
  {
    return y0 + v0 * t - 0.5 * g * t * t;
  }
  
  public double analyticVelocity (double v0)
  {
    return v0 - g * t;
  }
}
